modules/ocf_irc/manifests/webirc.pp

```diff
@@ -1,7 +1,9 @@
 class ocf_irc::webirc {
-  $webirc_fqdn = $::hostname ? {
-    /^dev-/ => 'dev-irc.ocf.berkeley.edu',
-    default => 'irc.ocf.berkeley.edu',
+  if $::dev_config {
+    $webirc_fqdn = 'dev-irc.ocf.berkeley.edu'
+  }
+  else {
+    $webirc_fqdn = 'irc.ocf.berkeley.edu'
```

Comment:
a slightly more canonical way to write this in puppet is:

```puppet
$webirc_fqdn = $::dev_config ? {
  true  => 'dev-irc.ocf.berkeley.edu',
  false => 'irc.ocf.berkeley.edu',
}
```

this makes it clear that the variable will *always* have a value, and makes it easier to avoid mistakes when refactoring (e.g. accidentally not setting the variable in one of the conditions)

Reply:
I think giving a `default` value here (instead of `false`) is even better to make sure that it always has a value, so that if it happens to not be boolean for some reason that it still gets a value.

Reply:
i think it's actually better to avoid default when you expect it to always be one of the two values, because it will error (and make it clear that something is wrong with your inputs) if one of the arms doesn't match:

```puppet
ckuehl@ithuriel:~$ cat test.pp  
$foo = 'blah'
$x = $foo ? {
  true => 1,
  false => 2,
}
```
```bash
ckuehl@ithuriel:~$ puppet apply test.pp
Error: Evaluation Error: No matching entry for selector parameter with value 'blah' at /home/ckuehl/test.pp:2:6 on node ithuriel.home.ckuehl.me

```